clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java

```diff
@@ -117,9 +116,9 @@ public void testEnsureCloseStopsRunningThread() {
     @ParameterizedTest
     @ValueSource(longs = {ConsumerNetworkThread.MAX_POLL_TIMEOUT_MS - 1, ConsumerNetworkThread.MAX_POLL_TIMEOUT_MS, ConsumerNetworkThread.MAX_POLL_TIMEOUT_MS + 1})
     public void testConsumerNetworkThreadPollTimeComputations(long exampleTime) {
-        List<Optional<? extends RequestManager>> list = new ArrayList<>();
-        list.add(Optional.of(coordinatorRequestManager));
-        list.add(Optional.of(heartbeatRequestManager));
+        List<RequestManager> list = new ArrayList<>();
```

Comment:
Could we use `List.of` here?

Reply:
Not easily, no. I made two different attempts (see commit history), and both just added a bunch of ugly code just to make this look cleaner.

The problem is that the list contains both `RequestManager`s and `Optional<RequestManager>`s, so the type of the list would have to be defined as `List<Object>`. The idea of this PR is to make the `entries` list only include the succinct set of `RequestManager`s that are actually in use.

I'm open to suggestions, though 😄

Reply:
@kirktrue I think I'm missing something here. Now that the list only contains `RequestManager` instances, can't you just write:
```
List<RequestManager> list = List.of(coordinatorRequestManager, heartbeatRequestManager);
```

I tried that locally and it looked good to me.

Reply:
Oof! I was looking at the lists being built up in `RequestManagers`’ constructors 🤦‍♂️ I've updated `ConsumerNetworkThreadTest`’s list building to use `List.of()` as suggested.